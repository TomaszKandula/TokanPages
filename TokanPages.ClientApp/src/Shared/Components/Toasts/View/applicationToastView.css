.toast-view {
    width: 100%;
}
