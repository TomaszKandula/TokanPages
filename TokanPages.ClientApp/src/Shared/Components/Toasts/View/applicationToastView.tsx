import React from "react";
import Snackbar from "@material-ui/core/Snackbar";
import { SlideProps, SnackbarOrigin } from "@material-ui/core";
import { Alert, Color } from "@material-ui/lab";
import { ReactSyntheticEvent } from "../../../../Shared/types";

interface Properties {
    anchorOrigin: SnackbarOrigin;
    isOpen: boolean;
    autoHideDuration: number;
    closeEventHandler: (event?: ReactSyntheticEvent, reason?: string) => void;
    TransitionComponent: (props: Omit<SlideProps, "direction">) => React.ReactElement;
    componentKey: string;
    toastSeverity: Color;
    toastMessage: string;
}

export const ApplicationToastView = (props: Properties): React.ReactElement => {
    return (
        <div className="toast-view">
            <Snackbar
                anchorOrigin={props.anchorOrigin}
                open={props.isOpen}
                autoHideDuration={props.autoHideDuration}
                onClose={props.closeEventHandler}
                TransitionComponent={props.TransitionComponent}
                key={props.componentKey}
            >
                <Alert variant="filled" elevation={6} onClose={props.closeEventHandler} severity={props.toastSeverity}>
                    {props.toastMessage}
                </Alert>
            </Snackbar>
        </div>
    );
};
