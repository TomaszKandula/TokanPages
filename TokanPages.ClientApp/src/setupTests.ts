// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import "@testing-library/jest-dom";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import enableHooks from "jest-react-hooks-shallow";

enableHooks(jest);

jest.mock("react-redux", () => (
{
    useSelector: jest.fn(),
    useDispatch: jest.fn()
}));

jest.mock("connected-react-router", () => (
{
    connectRouter: jest.fn(),
    routerMiddleware: jest.fn()
}));

Enzyme.configure({ adapter: new Adapter(), disableLifecycleMethods: false });

// environmental variables used during all tests
process.env.REACT_APP_API_VER="1";
process.env.REACT_APP_BACKEND="http://localhost:5000";
