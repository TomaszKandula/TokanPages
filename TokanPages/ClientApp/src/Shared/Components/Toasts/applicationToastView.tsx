import React from "react";
import Snackbar from "@material-ui/core/Snackbar";
import { SnackbarOrigin } from "@material-ui/core";
import { Alert, Color } from "@material-ui/lab";
import applicationToastViewStyle from "./applicationToastViewStyle";

interface IBinding
{
    bind: IProperties;
}

interface IProperties
{
    anchorOrigin: SnackbarOrigin;
    isOpen: boolean;
    autoHideDuration: number;
    closeEventHandler: any;
    TransitionComponent: any;
    key: any;
    toastSeverity: Color;
    toastMessage: string;
}

export default function ApplicationToastView(props: IBinding) 
{
    const classes = applicationToastViewStyle();
    return (
        <div className={classes.root}>
            <Snackbar 
                anchorOrigin={props.bind?.anchorOrigin} 
                open={props.bind?.isOpen} 
                autoHideDuration={props.bind?.autoHideDuration}
                onClose={props.bind?.closeEventHandler} 
                TransitionComponent={props.bind?.TransitionComponent} 
                key={props.bind?.key}
            >
                <Alert onClose={props.bind?.closeEventHandler} severity={props.bind?.toastSeverity} elevation={6} variant="filled">
                    {props.bind?.toastMessage}
                </Alert>
            </Snackbar>
        </div>
    );
}
