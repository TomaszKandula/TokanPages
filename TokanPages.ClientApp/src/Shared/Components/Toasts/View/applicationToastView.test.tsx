import "../../../../setupTests";
import React from "react";
import { render } from "@testing-library/react";
import { ApplicationToastView } from "./applicationToastView";
import { Slide, SlideProps } from "@material-ui/core";

describe("test view component for application toast", () => {
    it("should render correctly view component with passed props.", () => {
        const vertical = "top";
        const horizontal = "right";
        const toastSeverity = "error";
        const autoHideDuration: number = 15000;

        const TransitionLeft = (props: Omit<SlideProps, "direction">) => <Slide {...props} direction="left" />;

        const html = render(
            <ApplicationToastView
                anchorOrigin={{ vertical, horizontal }}
                isOpen={false}
                autoHideDuration={autoHideDuration}
                closeEventHandler={jest.fn()}
                TransitionComponent={TransitionLeft}
                componentKey={vertical + horizontal}
                toastSeverity={toastSeverity}
                toastMessage={"Test error message"}
            />
        );

        expect(html).toMatchSnapshot();
    });
});
