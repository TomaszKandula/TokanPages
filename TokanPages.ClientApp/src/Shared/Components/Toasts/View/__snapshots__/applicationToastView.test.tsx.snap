// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test view component for application toast should render correctly view component with passed props. 1`] = `
<div
  class="makeStyles-root-1"
/>
`;
